### PR TITLE
🧹 Configure Renovate

### LINKED ISSUE #3: Renovate Config
Most repositories for our Wayfair OSS offering are using either Dependabot or Renovate to maintain dependencies. We'd like to use renovate only as a solution moving forward, so we ought to [copy the config file](https://github.com/wayfair-incubator/forker/blob/98d1af861af424e45a1659eeec49fa59c8a6a846/renovate.json) (perhaps updating to give a weekly digest of changes) and merge in to provide automated dependency updates.

### PR DESCRIPTION
**Pull request by lelia**
Fixes #3. 

**Question for discussion:** Do we only want to support auto-merging PRs for minor/patch updates, or do we want to enable this for major dependency upgrades (eg. for Gatsby/React), too?

# Changes

* Adds `renovate.json` configuration file to `auxeng-docs`
* Scheduled to run on weekly basis (`earlyMondays`)
* Groups together updates for `gatsby` and `react`
* Supports both `npm` and `github-actions` updates
* Currently set to `automerge` minor/patch updates
* Adds `.github/lint.yml` workflow to lint config changes

### CODE REVIEW

File: .github/workflows/spellcheck.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Spellcheck
+name: spellcheck
```
Comment (GaryPWhite) [inline .github/workflows/spellcheck.yml:1]: is this a convention to name jobs lowercase?

File: renovate.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      "matchPackageNames": ["react"],
+      "matchPackagePrefixes": ["react-"],
+      "groupName": "React"
+    },
+    {
+      "matchManagers": ["github-actions"],
+      "groupName": "GitHub Actions"
+    }
+  ]
+}
```
Comment (GaryPWhite) [inline renovate.json:37]: looks awesome~!

**Review by GaryPWhite — APPROVED** [review]
some comments but generally lgtm

**Comment by lelia** [comment]
@GaryPWhite what are your thoughts on this Q from the description?

>Question for discussion: Do we only want to support auto-merging PRs for minor/patch updates, or do we want to enable this for major dependency upgrades (eg. for Gatsby/React), too?

File: .github/workflows/validate.yml [hunk 3]
```diff
@@ -7,12 +7,10 @@ jobs:
   validate-build:
     runs-on: ubuntu-latest
     container:
-      image: node:16-alpine
+      image: node:16-alpine3.15
```
[the remark this answers is not in the document]
Comment (lelia) [reply .github/workflows/validate.yml:10]: Ah yeah, i only did this to get exact parity with the version specified in the new docker container. it probably doesnt _really_ make a difference in CI, more so did it for consistency's sake so if we ever move to a new docker image we can do a singular find+replace

[the remark this answers is not in the document]
Comment (GaryPWhite) [reply .github/workflows/validate.yml:10]: lit

**Comment by GaryPWhite** [comment]
Minor updates pls, major stuff is likely to break unexpectedly and I'd rather have the chance to poke at it. Being a UI-mostly app, it's not easy to build a testing framework to make sure it looks / builds right with github actions.

**Comment by lelia** [comment]
> Minor updates pls, major stuff is likely to break unexpectedly and I'd rather have the chance to poke at it. Being a UI-mostly app, it's not easy to build a testing framework to make sure it looks / builds right with github actions.

Sweet, I _believe_ with the way it's configured right now (`"matchUpdateTypes": ["minor", "patch", "pin", "digest"],`) that should achieve what we want. We can validate that theory once it's merged into `main` and we've given Renovate some time to do its thing. 🤞

File: .github/workflows/spellcheck.yml (hunk 1 above)
Comment (lelia) [reply .github/workflows/spellcheck.yml:1]: Not an official convention so much as to make it consistent across workflows. Right now there's a mixture of title case, lowercase, and kebab case in the current "Actions" view. If we want to converge on title casing in the future we can always throw a quick PR together.

**Pull request merged**